src/value.mts

```diff
@@ -720,77 +750,86 @@ export class ObjectValue extends Value {
   }
 
   // NON-SPEC
-  mark(m) {
+  mark(m: GCMarker) {
     m(this.properties);
     this.internalSlotsList.forEach((s) => {
+      // @ts-ignore
       m(this[s]);
     });
   }
 }
 
 export class ReferenceRecord {
+  Base: 'unresolvable' | Value;
+  ReferencedName: StringValue | SymbolValue | PrivateName;
+  Strict: BooleanValue;
+  ThisValue: ObjectValue | undefined;
   constructor({
     Base,
     ReferencedName,
     Strict,
     ThisValue,
-  }) {
+  }: Pick<ReferenceRecord, 'Base' | 'ReferencedName' | 'Strict' | 'ThisValue'>) {
     this.Base = Base;
     this.ReferencedName = ReferencedName;
     this.Strict = Strict;
     this.ThisValue = ThisValue;
   }
 
   // NON-SPEC
-  mark(m) {
+  mark(m: GCMarker) {
     m(this.Base);
     m(this.ReferencedName);
     m(this.ThisValue);
   }
 }
 
-export function Descriptor(O) {
-  if (new.target === undefined) {
-    return new Descriptor(O);
+// @ts-expect-error
+export function Descriptor(O: Pick<Descriptor, 'Configurable' | 'Enumerable' | 'Get' | 'Set' | 'Value' | 'Writable'>): Descriptor
+// @ts-expect-error
+export @callable() class Descriptor {
+  readonly Value?: Value;
+  readonly Get?: FunctionObjectValue;
+  readonly Set?: FunctionObjectValue;
+  readonly Writable?: BooleanValue;
+  readonly Enumerable?: BooleanValue;
+  readonly Configurable?: BooleanValue;
+  constructor(O: Pick<Descriptor, 'Configurable' | 'Enumerable' | 'Get' | 'Set' | 'Value' | 'Writable'>) {
+    this.Value = O.Value;
+    this.Get = O.Get;
+    this.Set = O.Set;
+    this.Writable = O.Writable;
+    this.Enumerable = O.Enumerable;
+    this.Configurable = O.Configurable;
+  }
+  everyFieldIsAbsent() {
```

Comment:
whatever eslint rule is missing that ensures there would be newlines between class members, could you please enable it